src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,139 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (post: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [inputs, setInputs] = useState({
+    title: '',
+    description: '',
+    imdbId: '',
+    imgUrl: '',
+    imdbUrl: '',
+    isImgUrlValid: true,
+    isImdbUrlValid: true,
+  });
+
+  const {
+    title,
+    description,
+    imdbId,
+    imgUrl,
+    imdbUrl,
+    isImgUrlValid,
+    isImdbUrlValid,
+  } = inputs;
+
+  const isDisabled = !title.trim()
+    || !imgUrl.trim()
+    || !imdbUrl.trim()
+    || !imdbId.trim()
+    || !isImgUrlValid
+    || !isImdbUrlValid;
+
+  // eslint-disable-next-line max-len
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+  const handleImgUrlChange = (event: string) => {
+    setInputs({
+      ...inputs,
+      imgUrl: event,
+      isImgUrlValid: pattern.test(event.trim()),
+    });
```

Comment:
```suggestion
    setInputs(currentInputs => ({
      ...currentInputs,
      imgUrl: event,
      isImgUrlValid: pattern.test(event.trim()),
    }));
```
in such cases, it's better to use the previous state value from the function argument. Read more: [useState](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state)